modules/terraform/validate_struct.go

```diff
@@ -0,0 +1,99 @@
+package terraform
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terratest/modules/collections"
+)
+
+type ValidationOptions struct {
+	RootDir string
+	//Sub-directories relative to the RootDir to walk recursively for all Terraform modules. For example, if your
+	// RootDir is /home/project and you want to validate all modules in home/project/terraform then pass "terraform"
+	// in TargetSubDirs
+	TargetSubDirs []string
+	//ExcludeDirs must be relative to the RootDir. For example if your RootDir is home/project and you want to exclude
+	// the path /home/project/test, pass "test" in ExcludeDirs
+	ExcludeDirs []string
+}
+
+// NewValidationOptions returns a ValidationOptions struct, with override-able sane defaults
+func NewValidationOptions(rootDir string, targetSubDirs, excludeDirs []string) (ValidationOptions, error) {
+	vo := ValidationOptions{
+		ExcludeDirs: []string{},
+	}
+
+	if rootDir == "" {
+		return vo, ValidationUndefinedRootDirErr{}
+	}
+	vo.RootDir = rootDir
+
+	// If no target sub directories are passed, default to recursively searching "modules" and "examples"
+	if len(targetSubDirs) == 0 {
+		vo.TargetSubDirs = []string{
+			"modules",
+			"examples",
+		}
```

Comment:
In general, I prefer explicit over implicit. In that world, would be better to throw an error here.

From there, either provide a global var that includes these two that users can use (e.g., `DefaultTargetSubDirs`), or a separate function (e.g., `NewValidationOptionsWithDefaults`).

Reply:
I don't think we need to target `modules` and `examples` at all. I think the default should be to find all `.tf` files and run `validate` in the corresponding folders. If someone wants to target specific subfolders, then we can support include/exclude semantics in `Options`.